pybedlite/bed_record.py

```diff
@@ -11,12 +11,20 @@
     - :class:`~pybedtools.bed_record.BedRecord` -- Lightweight class for storing information
         pertaining to a BED record.
 """
+
+from __future__ import annotations
+
 import attr
 import enum
 from typing import Optional
 from typing import Tuple
 from typing import List
 from typing import ClassVar
+from typing import Type
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from pybedlite.overlap_detector import Interval
```

Comment:
@nh13 is this an acceptable workaround to solve the circular import?

Reply:
(see also the `from __future__ import annotations` above - I haven't seen this import used much in Fulcrum repos and am not sure if there are institutional objections)

Reply:
That's the way I've seen it done